releasenotes/notes/add-prettyprint-8ec601f41c9cc7d2.yaml

```
@@ -0,0 +1,86 @@
+---
+upgrade:
+  - |
+    Shows a warning message if non-printable strings are set to
+    :class:`~qiskit_optimization.QuadraticProgram`
+    as problem name, variable name, or constraint name.
+  - |
+    If users set an empty variable name ``""`` with
+    :meth:`~qiskit_optimization.QuadraticProgram.binary_var`,
+    :meth:`~qiskit_optimization.QuadraticProgram.integer_var`, and
+    :meth:`~qiskit_optimization.QuadraticProgram.continuous_var`,
+    they set the default variable name (e.g., ``x0``)
+    while they used to set the empty name as variable name.
+
+    .. code-block:: python
+
+        from qiskit_optimization.problems import QuadraticProgram
+
+        qp = QuadraticProgram()
+        x = qp.binary_var(name="")
+        y = qp.integer_var(name="")
+        z = qp.continuous_var(name="")
+        print(x.name)  # x0
+        print(y.name)  # x1
+        print(z.name)  # x2
+
+features:
+  - |
+    Adds a method :meth:`~qiskit_optimization.QuadraticProgram.prettyprint`
+    to ``QuadraticProgram`` to generate a pretty-printed string of a ``QuadraticProgram`` object.
```

```suggestion
    to :class:`~qiskit_optimization.QuadraticProgram` to generate a pretty-printed string of the object.
```
With the double single quotes its highlighted and looks like a link but is not, this makes it a link so the user can click on it if they want to go to the class itself.